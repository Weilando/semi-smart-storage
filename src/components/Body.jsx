import React from 'react';
import PropTypes from 'prop-types';
import StorageView from './StorageView';
import StorageSelector from './StorageSelector';
import '../styles/Body.css';

function Body(props) {
  Body.propTypes = {
    currentStorage: PropTypes.shape(
      {id: PropTypes.number.isRequired, name: PropTypes.string.isRequired}
    ).isRequired,
    currentStorageContent: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number.isRequired,
        item: PropTypes.string.isRequired,
        unit: PropTypes.string.isRequired,
        quantity: PropTypes.number.isRequired
      }).isRequired
    ).isRequired,
    storageList: PropTypes.arrayOf(
      PropTypes.shape(
        {id: PropTypes.number.isRequired, name: PropTypes.string.isRequired}
      )
    ).isRequired,
    itemList: PropTypes.arrayOf(
      PropTypes.shape(
        {id: PropTypes.number.isRequired, name: PropTypes.string.isRequired}
      )
    ).isRequired,
    unitList: PropTypes.arrayOf(
      PropTypes.shape(
        {id: PropTypes.number.isRequired, name: PropTypes.string.isRequired}
      )
    ).isRequired,
    addAction: PropTypes.func.isRequired,
    deleteAction: PropTypes.func.isRequired,
    switchStorageAction: PropTypes.func.isRequired,
    updateAction: PropTypes.func.isRequired,
  }

  return (
    <div className="Body">
      <StorageSelector
        currentStorage={props.currentStorage}
        storageList={props.storageList}
        addAction={props.addAction}
        deleteAction={props.deleteAction}
        switchAction={props.switchStorageAction}
        updateAction={props.updateAction}
      />
      <StorageView
        storage={props.currentStorage}
        storageContent={props.currentStorageContent}
        itemList={props.itemList}
        unitList={props.unitList}
        addAction={props.addAction}
        deleteAction={props.deleteAction}
        updateAction={props.updateAction}
      />
    </div>
  );
}

export default Body;
