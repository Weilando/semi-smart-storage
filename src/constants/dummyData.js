const DummyItemList = [
  {id: 0, name: "Sparkling water"},
  {id: 1, name: "Apple juice"},
  {id: 2, name: "Craft beer"},
  {id: 3, name: "Milk"},
  {id: 5, name: "Coke"}
];

const DummyItemListShort = [
  {id: 0, name: "Sparkling water"},
  {id: 1, name: "Apple juice"},
  {id: 3, name: "Milk"}
];

const DummyStorageContent = [
  {id: 0, item: "Milk", unit: "1L", quantity: 2.7},
  {id: 1, item: "Coke", unit: "0.33L", quantity: 5},
  {id: 2, item: "Apple juice", unit: "0.5L", quantity: 2},
  {id: 4, item: "Craft Beer", unit: "0.33L", quantity: 6},
  {id: 5, item: "Sparkling water", unit: "0.75L", quantity: 21}
];

const DummyStorageContentShort = [
  {id: 0, item: "Milk", unit: "1L", quantity: 2.7},
  {id: 2, item: "Sparkling water", unit: "0.75L", quantity: 21}
];

const DummyStorageList = [
  {id: 0, name: 'Fridge'},
  {id: 1, name: 'Basement'},
  {id: 3, name: 'Garage'},
];

const DummyStorageListShort = [
  {id: 0, name: 'Basement'},
  {id: 3, name: 'Fridge'},
];

const DummyUnitList = [
  {id: 0, name: "0.33L"},
  {id: 1, name: "0.5L"},
  {id: 2, name: "0.75L"},
  {id: 3, name: "1.0L"}
];

const DummyUnitListShort = [
  {id: 0, name: "0.33L"},
  {id: 2, name: "0.75L"},
  {id: 3, name: "1.0L"}
];

export { DummyItemList, DummyStorageContent, DummyStorageList, DummyUnitList };
export { DummyItemListShort, DummyStorageContentShort, DummyStorageListShort, DummyUnitListShort };
