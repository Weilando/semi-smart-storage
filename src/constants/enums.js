// api-operations
const AddMode = {
  ITEM_LIST: 0, STORAGE_LIST: 1, STORAGE_CONTENT: 2, UNIT_LIST: 3
}

const DeleteMode = {
  ITEM_LIST: 0, STORAGE_LIST: 1, STORAGE_CONTENT: 2, UNIT_LIST: 3
}

const FetchMode = {
  ITEM_LIST: 0, STORAGE_LIST: 1, STORAGE_CONTENT: 2, UNIT_LIST: 3
}

const UpdateMode = {
  ITEM_LIST: 0,
  STORAGE_LIST: 1,
  STORAGE_CONTENT_DECREMENT: 2,
  STORAGE_CONTENT_INCREMENT: 3,
  STORAGE_CONTENT_EDIT: 4,
  UNIT_LIST: 5
}

// modal-modes
const ModalMode = {
  ITEM: "item",
  STORAGE: "storage",
  UNIT: "unit"
}

export { AddMode, DeleteMode, FetchMode, UpdateMode, ModalMode };
