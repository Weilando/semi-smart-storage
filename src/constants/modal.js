let ModalMode = {
  ITEM: "item",
  STORAGE: "storage",
  UNIT: "unit"
}

export { ModalMode }
