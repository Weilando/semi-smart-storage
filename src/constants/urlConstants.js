const BASEURL = "http://raspberrypi/api/";
const ADD = "add.php";
const DELETE = "delete.php";
const GET = "get.php";
const UPDATE = "update.php";

export { BASEURL };
export { ADD, DELETE, GET, UPDATE };
