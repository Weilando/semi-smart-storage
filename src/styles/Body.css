div.Body {
  background-color: lightgrey;
  height: 95vh;
  width: 100vw;
  min-height: 80px;
  min-width: 100px;
  margin: 0;
  padding: 0;
}
