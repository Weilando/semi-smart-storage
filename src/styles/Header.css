div.Header {
  background-color: grey;
  height: 5vh;
  width: 100vw;
  min-height: 10px;
  min-width: 100px;
  padding: 1em;
  margin: 0;
}
