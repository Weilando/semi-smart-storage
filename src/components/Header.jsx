import React from 'react';
import PropTypes from 'prop-types';
import ModalEdit from './ModalEdit';
import ModalSettings from './ModalSettings';
import ModalList from './ModalList';
import { AddMode, DeleteMode, UpdateMode } from '../constants/enums';
import { ModalMode } from '../constants/enums';
import '../styles/Header.css';

class Header extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      currentMode: ModalMode.ITEM,
      settingsShow: false,
      unitListShow: false,
      itemListShow: false,
      editShow: false,
      editId: 0,
      editName: '',
      editBuffer: '',
    }

    this.closeEditModal = this.closeEditModal.bind(this);
    this.closeItemListModal = this.closeItemListModal.bind(this);
    this.closeUnitListModal = this.closeUnitListModal.bind(this);
    this.closeSettingsModal = this.closeSettingsModal.bind(this);
    this.generateEditModal = this.generateEditModal.bind(this);
    this.generateListModal = this.generateListModal.bind(this);
    this.generateSettingsModal = this.generateSettingsModal.bind(this);
    this.showEditModal = this.showEditModal.bind(this);
    this.showItemListModal = this.showItemListModal.bind(this);
    this.showUnitListModal = this.showUnitListModal.bind(this);
    this.showSettingsModal = this.showSettingsModal.bind(this);
    this.handleEditChange = this.handleEditChange.bind(this);
    this.handleItemAdd = this.handleItemAdd.bind(this);
    this.handleUnitAdd = this.handleUnitAdd.bind(this);
    this.handleItemDelete = this.handleItemDelete.bind(this);
    this.handleUnitDelete = this.handleUnitDelete.bind(this);
    this.handleItemUpdate = this.handleItemUpdate.bind(this);
    this.handleUnitUpdate = this.handleUnitUpdate.bind(this);
  }

  static propTypes = {
    itemList: PropTypes.arrayOf(
      PropTypes.shape(
        {id: PropTypes.number.isRequired, name: PropTypes.string.isRequired}
      )
    ).isRequired,
    unitList: PropTypes.arrayOf(
      PropTypes.shape(
        {id: PropTypes.number.isRequired, name: PropTypes.string.isRequired}
      )
    ).isRequired,
    dummy: PropTypes.bool.isRequired,
    addAction: PropTypes.func.isRequired,
    deleteAction: PropTypes.func.isRequired,
    reloadAction: PropTypes.func.isRequired,
    switchDummyAction: PropTypes.func.isRequired,
    updateAction: PropTypes.func.isRequired,
  }


  // modal-handling
  showEditModal(obj) {
    if(this.state.editShow === false) {
      this.setState({editShow: true, editId: obj.id, editName: obj.name, editBuffer: obj.name});
    }
  }

  closeEditModal() {
    if(this.state.editShow === true) {
      this.setState({editShow: false});
    }
  }

  showItemListModal() {
    if(this.state.itemListShow === false) {
      this.setState({itemListShow: true, currentMode: ModalMode.ITEM});
    }
  }

  closeItemListModal() {
    if(this.state.itemListShow === true) {
      this.setState({itemListShow: false});
    }
  }

  showUnitListModal() {
    if(this.state.unitListShow === false) {
      this.setState({unitListShow: true, currentMode: ModalMode.UNIT});
    }
  }

  closeUnitListModal() {
    if(this.state.unitListShow === true) {
      this.setState({unitListShow: false});
    }
  }

  showSettingsModal() {
    if(this.state.settingsShow === false) {
      this.setState({settingsShow: true});
    }
  }

  closeSettingsModal() {
    if(this.state.settingsShow === true) {
      this.setState({settingsShow: false});
    }
  }


  // action handling
  handleEditChange(event) {
    this.setState({editBuffer: event.target.value});
  }

  handleNewNameChange(event) {
    this.setState({newNameBuffer: event.target.value});
  }

  handleItemAdd(name) {
    this.props.addAction(AddMode.ITEM_LIST, name);
  }

  handleItemUpdate(event) {
    event.preventDefault();
    if(this.state.editBuffer.localeCompare(this.state.editName) !== 0) {
      this.props.updateAction(UpdateMode.ITEM_LIST, this.state.editId, this.state.editBuffer);
    }
    this.closeEditModal();
  }

  handleItemDelete() {
    this.props.deleteAction(DeleteMode.ITEM_LIST, this.state.editId);
    this.closeEditModal();
  }

  handleUnitAdd(name) {
    this.props.addAction(AddMode.UNIT_LIST, name);
  }

  handleUnitUpdate(event) {
    event.preventDefault();
    if(this.state.editBuffer.localeCompare(this.state.editName) !== 0) {
      this.props.updateAction(UpdateMode.UNIT_LIST, this.state.editId, this.state.editBuffer);
    }
    this.closeEditModal();
  }

  handleUnitDelete() {
    this.props.deleteAction(DeleteMode.UNIT_LIST, this.state.editId);
    this.closeEditModal();
  }


  // rendering
  generateEditModal() {
    return(
      <ModalEdit
        show={this.state.editShow}
        name={this.state.editName}
        mode={this.state.currentMode}
        buffer={this.state.editBuffer}
        changeAction={this.handleEditChange}
        closeAction={this.closeEditModal}
        deleteAction={(this.state.currentMode === ModalMode.UNIT) ? this.handleUnitDelete : this.handleItemDelete}
        updateAction={(this.state.currentMode === ModalMode.UNIT) ? this.handleUnitUpdate : this.handleItemUpdate}
      />
    );
  }

  generateListModal(mode) {
    return(
      <ModalList
        show={(mode === ModalMode.UNIT) ? this.state.unitListShow : this.state.itemListShow}
        mode={mode}
        list={(mode === ModalMode.UNIT) ? this.props.unitList : this.props.itemList}
        addAction={(mode === ModalMode.UNIT) ? this.handleUnitAdd : this.handleItemAdd}
        closeAction={(mode === ModalMode.UNIT) ? this.closeUnitListModal : this.closeItemListModal}
        editAction={this.showEditModal}
      />
    );
  }

  generateSettingsModal() {
    return(
      <ModalSettings
        dummy={this.props.dummy}
        show={this.state.settingsShow}
        closeAction={this.closeSettingsModal}
        switchDummyAction={this.props.switchDummyAction}
      />
    );
  }

  render() {
    const itemListModal = this.generateListModal(ModalMode.ITEM);
    const unitListModal = this.generateListModal(ModalMode.UNIT);
    const settingsModal = this.generateSettingsModal();
    const editModal = this.generateEditModal();

    return (
      <div className="Header">
        <button className="Header" onClick={this.props.reloadAction}>
          Reload
        </button>
        <button className="Header" onClick={this.showSettingsModal}>
          Settings
        </button>
        <button className="Header" onClick={this.showUnitListModal}>
          Units
        </button>
        <button className="Header" onClick={this.showItemListModal}>
          Items
        </button>

        <h1 className="Header">SemiSmartStorage</h1>

        {itemListModal}
        {unitListModal}
        {settingsModal}
        {editModal}
      </div>
    );
  }
}

export default Header;
