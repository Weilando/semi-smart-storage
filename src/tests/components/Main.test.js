import React from 'react';
import renderer from 'react-test-renderer';
import Body from '../../components/Body';
import Header from '../../components/Header';
import { DummyItemListShort, DummyStorageContentShort, DummyStorageListShort, DummyUnitListShort } from '../../constants/dummyData';

describe('snapshot tests for Body component', () => {
  it('should render Body component correctly', () => {
    const addStorageAction = jest.fn();
    const deleteStorageAction = jest.fn();
    const switchStorageAction = jest.fn();
    const updateStorageAction = jest.fn();
    const addItemToStorageAction = jest.fn();
    const decrementQuantityForItemAction = jest.fn();
    const incrementQuantityForItemAction = jest.fn();
    const removeItemFromStorageAction = jest.fn();

    const tree = renderer.create(
      <Body
        currentStorage={DummyStorageListShort[1]}
        currentStorageContent={DummyStorageContentShort}
        storageList={DummyStorageListShort}
        itemList={DummyItemListShort}
        unitList={DummyUnitListShort}
        addStorageAction={addStorageAction}
        deleteStorageAction={deleteStorageAction}
        switchStorageAction={switchStorageAction}
        updateStorageAction={updateStorageAction}
        addItemToStorageAction={addItemToStorageAction}
        decrementQuantityForItemAction={decrementQuantityForItemAction}
        incrementQuantityForItemAction={incrementQuantityForItemAction}
        removeItemFromStorageAction={removeItemFromStorageAction}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  })
});

describe('snapshot tests for Header component', () => {
  it('should render Header component correctly', () => {
    const reloadAction = jest.fn();
    const switchDummyAction = jest.fn();
    const addItemAction = jest.fn();
    const addUnitAction = jest.fn();
    const deleteItemAction = jest.fn();
    const deleteUnitAction = jest.fn();
    const updateItemAction = jest.fn();
    const updateUnitAction = jest.fn();

    const tree = renderer.create(
      <Header
        itemList={DummyItemListShort}
        unitList={DummyUnitListShort}
        dummy={true}
        reloadAction={reloadAction}
        switchDummyAction={switchDummyAction}
        addItemAction={addItemAction}
        addUnitAction={addUnitAction}
        deleteItemAction={deleteItemAction}
        deleteUnitAction={deleteUnitAction}
        updateItemAction={updateItemAction}
        updateUnitAction={updateUnitAction}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  })
});
