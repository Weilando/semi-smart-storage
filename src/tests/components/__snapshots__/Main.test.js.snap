// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot tests for Body component should render Body component correctly 1`] = `
<div
  className="Body"
>
  <div
    className="StorageSelector"
  >
    <ul
      className="StorageSelector"
    >
      <li
        className="StorageSelector"
      >
        <button
          className="Block"
          onClick={[Function]}
        >
          <p
            className=""
          >
            Basement
          </p>
        </button>
        <button
          className="Round"
          onClick={[Function]}
        >
          i
        </button>
      </li>
      <li
        className="StorageSelectorActive"
      >
        <button
          className="Block"
          onClick={[Function]}
        >
          <p
            className="Active"
          >
            Fridge
          </p>
        </button>
        <button
          className="Round"
          onClick={[Function]}
        >
          i
        </button>
      </li>
      <li
        className="StorageSelector"
      >
        <form
          onSubmit={[Function]}
        >
          <input
            className="StorageSelector"
            onChange={[Function]}
            type="text"
            value=""
          />
          <input
            className="StorageSelector"
            disabled={true}
            type="submit"
            value="Add"
          />
        </form>
      </li>
    </ul>
    <div
      className="modal hide"
    >
      <div
        className="modal-main"
      >
        <span
          className="close"
          onClick={[Function]}
        >
          ×
        </span>
        <h1>
          Settings for 
          storage
           "
          Fridge
          ".
        </h1>
        <h2>
          Update name
        </h2>
        <form
          className="Modal"
          onSubmit={[Function]}
        >
          <input
            className="Modal"
            onChange={[Function]}
            type="text"
            value=""
          />
          <br />
          <input
            className="Modal"
            disabled={false}
            type="submit"
            value="Update name"
          />
        </form>
        <h2>
          Delete 
          storage
        </h2>
        <p>
          To delete the storage and all its content, click the button below. Items and units are not deleted.
        </p>
        <button
          className="Modal"
          onClick={[Function]}
        >
          Delete 
          storage
        </button>
      </div>
    </div>
  </div>
  <div
    className="StorageView"
  >
    <table
      className="StorageView"
    >
      <thead>
        <tr>
          <th
            width="50%"
          >
            Name
          </th>
          <th
            width="20%"
          >
            Unit
          </th>
          <th
            width="20%"
          >
            Quantity
          </th>
          <th
            width="10%"
          >
            Options
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Milk
          </td>
          <td>
            1L
          </td>
          <td>
            2.7
          </td>
          <td>
            <button
              className="StorageView"
              disabled={false}
              onClick={[Function]}
            >
              -
            </button>
            <button
              className="StorageView"
              onClick={[Function]}
            >
              +
            </button>
            <button
              className="StorageView"
              onClick={[Function]}
            >
              Edit
            </button>
            <button
              className="StorageView"
              onClick={[Function]}
            >
              Remove
            </button>
          </td>
        </tr>
        <tr>
          <td>
            Sparkling water
          </td>
          <td>
            0.75L
          </td>
          <td>
            21
          </td>
          <td>
            <button
              className="StorageView"
              disabled={false}
              onClick={[Function]}
            >
              -
            </button>
            <button
              className="StorageView"
              onClick={[Function]}
            >
              +
            </button>
            <button
              className="StorageView"
              onClick={[Function]}
            >
              Edit
            </button>
            <button
              className="StorageView"
              onClick={[Function]}
            >
              Remove
            </button>
          </td>
        </tr>
      </tbody>
      <tfoot>
        <tr>
          <td>
            <select
              id="Item"
              name="Item"
              onChange={[Function]}
              value={-1}
            >
              <option
                className="choose"
                value={-1}
              >
                -Choose item-
              </option>
              <option
                value={0}
              >
                Sparkling water
              </option>
              <option
                value={1}
              >
                Apple juice
              </option>
              <option
                value={3}
              >
                Milk
              </option>
            </select>
          </td>
          <td>
            <select
              id="Unit"
              name="Unit"
              onChange={[Function]}
              value={-1}
            >
              <option
                className="choose"
                value={-1}
              >
                -Choose unit-
              </option>
              <option
                value={0}
              >
                0.33L
              </option>
              <option
                value={2}
              >
                0.75L
              </option>
              <option
                value={3}
              >
                1.0L
              </option>
            </select>
          </td>
          <td
            colSpan="2"
          >
            <form
              onSubmit={[Function]}
            >
              <input
                min="0"
                onChange={[Function]}
                step="0.05"
                type="number"
                value={0}
              />
              <input
                disabled={true}
                type="submit"
                value="Add"
              />
            </form>
          </td>
        </tr>
      </tfoot>
    </table>
    <div
      className="modal hide"
    >
      <div
        className="modal-main"
      >
        <span
          className="close"
          onClick={[Function]}
        >
          ×
        </span>
        <h1>
          Change quantity
        </h1>
        <form
          className="Modal"
          onSubmit={[Function]}
        >
          <input
            className="Modal"
            min="0"
            onChange={[Function]}
            step="0.05"
            type="number"
            value={0}
          />
          <input
            className="Modal"
            type="submit"
            value="Update quantity"
          />
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`snapshot tests for Header component should render Header component correctly 1`] = `
<div
  className="Header"
>
  <button
    className="Header"
    onClick={[MockFunction]}
  >
    Reload
  </button>
  <button
    className="Header"
    onClick={[Function]}
  >
    Settings
  </button>
  <button
    className="Header"
    onClick={[Function]}
  >
    Units
  </button>
  <button
    className="Header"
    onClick={[Function]}
  >
    Items
  </button>
  <h1
    className="Header"
  >
    SemiSmartStorage
  </h1>
  <div
    className="modal hide"
  >
    <div
      className="modal-main"
    >
      <span
        className="close"
        onClick={[Function]}
      >
        ×
      </span>
      <h1>
        Settings for 
        item
      </h1>
      <p>
        Click the edit-button next to the 
        item
         you want to edit and perform your changes.
      </p>
      <ul
        className="Modal"
      >
        <li
          className="Modal"
        >
          <button
            className="Block"
          >
            <p>
              Sparkling water
            </p>
          </button>
          <button
            className="Round"
            onClick={[Function]}
          >
            i
          </button>
        </li>
        <li
          className="Modal"
        >
          <button
            className="Block"
          >
            <p>
              Apple juice
            </p>
          </button>
          <button
            className="Round"
            onClick={[Function]}
          >
            i
          </button>
        </li>
        <li
          className="Modal"
        >
          <button
            className="Block"
          >
            <p>
              Milk
            </p>
          </button>
          <button
            className="Round"
            onClick={[Function]}
          >
            i
          </button>
        </li>
        <li
          className="Modal"
        >
          <form
            onSubmit={[Function]}
          >
            <input
              className="ModalList"
              onChange={[Function]}
              type="text"
              value=""
            />
            <input
              className="ModalList"
              disabled={true}
              type="submit"
              value="Add"
            />
          </form>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="modal hide"
  >
    <div
      className="modal-main"
    >
      <span
        className="close"
        onClick={[Function]}
      >
        ×
      </span>
      <h1>
        Settings for 
        unit
      </h1>
      <p>
        Click the edit-button next to the 
        unit
         you want to edit and perform your changes.
      </p>
      <ul
        className="Modal"
      >
        <li
          className="Modal"
        >
          <button
            className="Block"
          >
            <p>
              0.33L
            </p>
          </button>
          <button
            className="Round"
            onClick={[Function]}
          >
            i
          </button>
        </li>
        <li
          className="Modal"
        >
          <button
            className="Block"
          >
            <p>
              0.75L
            </p>
          </button>
          <button
            className="Round"
            onClick={[Function]}
          >
            i
          </button>
        </li>
        <li
          className="Modal"
        >
          <button
            className="Block"
          >
            <p>
              1.0L
            </p>
          </button>
          <button
            className="Round"
            onClick={[Function]}
          >
            i
          </button>
        </li>
        <li
          className="Modal"
        >
          <form
            onSubmit={[Function]}
          >
            <input
              className="ModalList"
              onChange={[Function]}
              type="text"
              value=""
            />
            <input
              className="ModalList"
              disabled={true}
              type="submit"
              value="Add"
            />
          </form>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="modal hide"
  >
    <div
      className="modal-main"
    >
      <span
        className="close"
        onClick={[Function]}
      >
        ×
      </span>
      <h1>
        Settings
      </h1>
      <h2>
        Click-dummy
      </h2>
      <p>
        This option enables fake-data to showcase the UI. Neither an active network connection, nor a running backend are necessary. Changes are not persistent.
      </p>
      <button
        name="toggle"
        onClick={[MockFunction]}
      >
        Switch click-dummy off
      </button>
    </div>
  </div>
  <div
    className="modal hide"
  >
    <div
      className="modal-main"
    >
      <span
        className="close"
        onClick={[Function]}
      >
        ×
      </span>
      <h1>
        Settings for 
        item
         "
        
        ".
      </h1>
      <h2>
        Update name
      </h2>
      <form
        className="Modal"
        onSubmit={[Function]}
      >
        <input
          className="Modal"
          onChange={[Function]}
          type="text"
          value=""
        />
        <br />
        <input
          className="Modal"
          disabled={true}
          type="submit"
          value="Update name"
        />
      </form>
      <h2>
        Delete 
        item
      </h2>
      <p>
        To delete the item, click the button below. All occurances in every storage will be deleted, too.
      </p>
      <button
        className="Modal"
        onClick={[Function]}
      >
        Delete 
        item
      </button>
    </div>
  </div>
</div>
`;
